### ISSUE TITLE
Incremental learning with boosting from n_trees <= trees.size() (effectively abandoning trees > n_trees)

### ISSUE THREAD

**Issue by jokari69** [dmlc/xgboost#2281]
I created a callback to set the learning rates dynamically (cross over of early stop and reset learning rates)
Whenever i drop my learning rate i would like to start boosting from best_iteration again instead of adding more boosters at the end.
As far I understand the code base this seems currently not possible (please guide me if it is)

Possible solutions are:
- Change UpdateOneIter and BoostOneIter  (allow to use the callbacks to set the learning rates)
- Change Save and/or LoadModel to only save/load the trees until best_iteration (this would use early stopping and save/load model to do incremental learning and set the learning rates outside the cv/train method)

Any preference or other solutions?
